ee/control/notify_consumer_test.go

```diff
@@ -0,0 +1,224 @@
+package control
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/osquery"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.etcd.io/bbolt"
+)
+
+type notifierMock struct{ mock.Mock }
+
+func newNotifierMock() *notifierMock { return &notifierMock{} }
+
+func (nm *notifierMock) SendNotification(title, body string) error {
+	args := nm.Called(title, body)
+	return args.Error(0)
+}
+
+func TestUpdate_HappyPath(t *testing.T) {
+	t.Parallel()
+
+	db := setUpDb(t)
+	mockNotifier := newNotifierMock()
+	testNc := &NotificationConsumer{
+		db:              db,
+		runner:          mockNotifier,
+		logger:          log.NewNopLogger(),
+		notificationTtl: time.Hour * 1,
+	}
+
+	// Send one notification that we haven't seen before
+	testTitle := fmt.Sprintf("Test title @ %d - 9779e3b9-75e6-4b59-8ed6-5a2d6d2934ea", time.Now().UnixMicro())
+	testBody := fmt.Sprintf("Test body @ %d - 9779e3b9-75e6-4b59-8ed6-5a2d6d2934ea", time.Now().UnixMicro())
+	testNotifications := []notification{
+		{
+			Title: testTitle,
+			Body:  testBody,
+			UUID:  "9779e3b9-75e6-4b59-8ed6-5a2d6d2934ea",
+		},
+	}
+	testNotificationsRaw, err := json.Marshal(testNotifications)
+	require.NoError(t, err)
+	testNotificationsData := bytes.NewReader(testNotificationsRaw)
+
+	// Expect that the notifier is called once to send the one notification successfully
+	mockNotifier.On("SendNotification", testTitle, testBody).Return(nil)
+
+	// Call update and assert our expectations about sent notifications
+	err = testNc.Update(testNotificationsData)
+	require.NoError(t, err)
+	mockNotifier.AssertCalled(t, "SendNotification", testTitle, testBody)
```

Comment:
I think this code may only be asserting that mockNotifier was called _at least_ once. I think this will make sure it was only called exactly once:
```suggestion
	mockNotifier.On("SendNotification", testTitle, testBody).Return(nil).Once()

	// Call update and assert our expectations about sent notifications
	err = testNc.Update(testNotificationsData)
	require.NoError(t, err)
	mockNotifier.AssertCalled(t, "SendNotification", testTitle, testBody)
	mockNotifier.AssertExpectations(t)
```

.... and now that I scrolled further down I can see you are validating the number of calls in other tests =). So, you could just keep it consistent and do the same as you did in the other tests and `AssertNumberOfCalls(...)`

Reply:
Good call, I'll update!